sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/metrics/models/MetricsHelper.java

```diff
@@ -268,4 +316,60 @@ public static String getSubscriptionFromResourceId(String s) {
         return subscriptionId;
     }
 
+    /**
+     * Converts a {@link QueryTimeInterval} to ISO 8601 string format suitable for Azure Monitor Metrics API.
+     * For duration-only intervals, this method converts them to absolute start/end times based on current time.
+     *
+     * @param timeInterval The time interval to convert.
+     * @return ISO 8601 formatted string representation with absolute start/end times.
+     */
+    public static String toMetricsTimespan(QueryTimeInterval timeInterval) {
+        if (timeInterval == null) {
+            return null;
+        }
+
+        // If we have both start and end times, use them directly
+        if (timeInterval.getStartTime() != null && timeInterval.getEndTime() != null) {
```

Comment:
We should throw an exception if all 3 are set - start time, end time and duration.